### ISSUE TITLE
API for sharing albums?

### ISSUE THREAD

**Issue by EddyLB** [petitssuisses/piwigo-ShareAlbum#33]
Hi Arnaud,
Would it be possible to provide an [API](http://piwigo.org/demo/tools/ws.htm#top) allowing to share an album? 
For example, the [iOS app](https://github.com/Piwigo/Piwigo-Mobile) could detect the availability of your extension on a Piwigo server using `reflection.getMethodList` and then allow users to exploit your plugin remotely.
Cheers

**Comment by EddyLB** [comment]
With this API method, third party apps could also "hide" the virtual albums created by the plugin as the Piwigo server does it.

**Comment by petitssuisses** [comment]
Hi Eddy, 
Sorry for this long time reply. For sure it would be possible, but I have to enquire on the 'how', and did not take time up to know. Any idea of any other plugin/contributor who implemented it for the development of the IOS app (or other usage ?) so that I could have a look ?
Cheers
Arnaud

**Comment by EddyLB** [comment]
Hi Arnaud,
Thank for your reply. I have never coded a plugin for Piwigo but I guess that you should be able to add an [API](http://piwigo.org/demo/tools/ws.htm) as the [Cummunity plugin](https://github.com/plegall/Piwigo-community) by @plegall does it [here](https://github.com/plegall/Piwigo-community/blob/master/main.inc.php) starting line 287 (`_ws_` related calls I guess).

The iOS app calls the [Piwigo method](http://piwigo.org/demo/tools/ws.htm) `reflection.getMethodList` to check whether the Community method `community.session.getStatus` exists. If it does, it then calls it to retrieve the data necessary to manage a server set with the Community plugin.
Cheers

**Comment by EddyLB** [comment]
Hi Arnaud, 
I continue to receive requests from iOS app users…
Could you please raise the priority of this issue?

**Comment by petitssuisses** [comment]
Hi Eddy,
I'll check, sure it would be a nice feature in the iOS app.
Cheers

**Comment by petitssuisses** [comment]
@EddyLB is there any other known plugin which exposes its methods through ws calls ?

**Comment by EddyLB** [comment]
Yes @petitssuisses, Community does it (https://github.com/plegall/Piwigo-community).

**Comment by EddyLB** [comment]
Any progress?

**Comment by petitssuisses** [comment]
Hi
I've implemented ws methods, it's committed to the main branch, can you give it a try from the main branch directly ?

**Comment by EddyLB** [comment]
Thanks for the API methods

<img width="603" height="1311" alt="Image" src="https://github.com/user-attachments/assets/dc4a3134-43ea-48a8-b456-eb9c20ac4997" />
